packages/react-hooks/src/ConnectionProvider.tsx

```diff
@@ -15,21 +15,57 @@ import {
   addRecord,
 } from './recordUtils'
 
+export type useConnectionsOptions = {
+  excludedTypes?: [ConnectionType | string]
+  connectionState?: DidExchangeState
+}
+
 const ConnectionContext = createContext<RecordsState<ConnectionRecord> | undefined>(undefined)
 
-export const useConnections = () => {
+/**
+ * This method retrieves the connection context for the current agent.
+ * From this you can access all connection records for the agent.
+ * @param options options for useConnections hook, lets us filter out specific types and limit states
+ * @returns a connection context containing information about the agents connections
+ */
+export const useConnections = (options: useConnectionsOptions = {}) => {
   const connectionContext = useContext(ConnectionContext)
-  if (!connectionContext) {
-    throw new Error('useConnections must be used within a ConnectionContextProvider')
-  }
-  return connectionContext
+
+  let connections = connectionContext?.records
+  
+  connections = useMemo(() => {
+    if (!connections) {
+      throw new Error('useConnections must be used within a ConnectionContextProvider')
+    }
+    // do not filter if not filter options are provided to save on a loop
+    if (!options.connectionState && !options.excludedTypes) return connections
+
+    return connections.filter((record: ConnectionRecord) => {
+      // By default we include this connection
+      
+      // Filter by state (if connectionState is defined)
+      if (options.connectionState !== record.state) return false
+      
+      // Exclude records with certain connection types (if defined)
+      const recordTypes = record.connectionTypes as ConnectionType[] | string[] | null
```

Comment:
It could contains string and ConnectionType items at the same time:

```suggestion
      const recordTypes = record.connectionTypes as Array<ConnectionType | string> | null
```